io/eolearn/io/sentinelhub_process.py

```diff
@@ -542,7 +541,9 @@ def _extract_additional_features(
         self, eopatch: EOPatch, images: Iterable[np.ndarray], shape: tuple[int, ...]
     ) -> None:
         """Extracts additional features from response into an EOPatch"""
-        additional_data = cast(List[FeatureRenameSpec], self.additional_data)  # verified by `if` in _extract_data
+        additional_data = cast(
+            List[Tuple[FeatureType, str, str]], self.additional_data
+        )  # verified by `if` in _extract_data
```

Comment:
You could probably drop the `cast` part if you replace with:

```suggestion
        if self.additional_data is None:
            return None
```

Reply:
did something similar